flags.go

```diff
@@ -77,13 +82,40 @@ func (b csvFileBackend) Refresh() (map[string]Flag, time.Time, error) {
 }
 
 type Flag struct {
-	Name string
-	Rate float64
+	Name   string
+	Active bool
+	Rules  []Rule
+}
+
+type Rule interface {
+	Handle(ctx context.Context, props map[string]string) bool
+	onMatch() string
+	onMiss() string
+}
+
+type MatchListRule struct {
+	Property string
+	Values   []string
+	OnMatch  string `json:"on_match"`
```

Comment:
It seems a bit weird for both rule types to have `OnMatch` and `OnMiss`. Maybe instead we can factor that out?

```go
type RuleInfo struct {
   OnMatch string
   OnMiss string
   Rule Rule
}
```